nbformat.js

```diff
@@ -1,31 +1,92 @@
-var Validator = require('jsonschema').Validator;
+var Ajv = require('ajv');
+var JaySchema = require('jayschema');
+
 var v3Schema = require('./v3/nbformat.v3.schema.json');
 var v4Schema = require('./v4/nbformat.v4.schema.json');
 
+/**
+ * NotebookValidator inherits from JaySchema for verbosity and composes ajv for
+ * optional speed.
+ */
 function NotebookValidator() {
-    Validator.call(this);
-    this.addSchema(v3Schema, '/v3');
-    this.addSchema(v4Schema, '/v4');
+    JaySchema.call(this);
+
+    this.ajv = new Ajv();
+
+    this.ajv.addSchema(v3Schema, '/v3');
+    this.ajv.addSchema(v4Schema, '/v4');
+
+    this.register(v3Schema, '/v3');
+    this.register(v4Schema, '/v4');
```

Comment:
If we straddle, we should make sure that `register` gets overridden to call `JaySchema.register.call(this...)` followed by `this.ajv.addSchema...`.